src/libraries/System.IO.FileSystem/tests/File/GetSetTimes.cs

```diff
@@ -62,74 +86,59 @@ public async Task CreationTimeSet_GetReturnsExpected_WhenNotInFuture()
                     DateTimeKind.Unspecified);
                 yield return TimeFunction.Create(
-                    ((path, time) => File.SetCreationTimeUtc(path, time)),
-                    ((path) => File.GetCreationTimeUtc(path)),
+                    SetCreationTimeUtc,
+                    GetCreationTimeUtc,
                     DateTimeKind.Utc);
             }
             yield return TimeFunction.Create(
-                ((path, time) => File.SetLastAccessTime(path, time)),
-                ((path) => File.GetLastAccessTime(path)),
+                SetLastAccessTime,
+                GetLastAccessTime,
                 DateTimeKind.Local);
             yield return TimeFunction.Create(
-                ((path, time) => File.SetLastAccessTimeUtc(path, time)),
-                ((path) => File.GetLastAccessTimeUtc(path)),
+                SetLastAccessTimeUtc,
+                GetLastAccessTimeUtc,
                 DateTimeKind.Unspecified);
             yield return TimeFunction.Create(
-                ((path, time) => File.SetLastAccessTimeUtc(path, time)),
-                ((path) => File.GetLastAccessTimeUtc(path)),
+                SetLastAccessTimeUtc,
+                GetLastAccessTimeUtc,
                 DateTimeKind.Utc);
             yield return TimeFunction.Create(
-                ((path, time) => File.SetLastWriteTime(path, time)),
-                ((path) => File.GetLastWriteTime(path)),
+                SetLastWriteTime,
+                GetLastWriteTime,
                 DateTimeKind.Local);
             yield return TimeFunction.Create(
-                ((path, time) => File.SetLastWriteTimeUtc(path, time)),
-                ((path) => File.GetLastWriteTimeUtc(path)),
+                SetLastWriteTimeUtc,
+                GetLastWriteTimeUtc,
                 DateTimeKind.Unspecified);
             yield return TimeFunction.Create(
-                ((path, time) => File.SetLastWriteTimeUtc(path, time)),
-                ((path) => File.GetLastWriteTimeUtc(path)),
+                SetLastWriteTimeUtc,
+                GetLastWriteTimeUtc,
                 DateTimeKind.Utc);
         }
 
-        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotInAppContainer))] // Can't read root in appcontainer
-        [PlatformSpecific(TestPlatforms.Windows)]
-        public void PageFileHasTimes()
-        {
-            // Typically there is a page file on the C: drive, if not, don't bother trying to track it down.
-            string pageFilePath = Directory.EnumerateFiles(@"C:\", "pagefile.sys").FirstOrDefault();
-            if (pageFilePath != null)
-            {
-                Assert.All(TimeFunctions(), (item) =>
-                {
-                    var time = item.Getter(pageFilePath);
-                    Assert.NotEqual(DateTime.FromFileTime(0), time);
-                });
-            }
-        }
-
         [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
```

Comment:
```
    System.IO.Tests.GetSetAttributes_SafeFileHandle.SettingAttributes_Unix(attributes: ReadOnly) [FAIL]
      System.UnauthorizedAccessException : Access to the path '/tmp/#GetSetAttributes_SafeFileHandle_omr41eqf.jtf/SettingAttributes_Unix_17_mfwmbgd3' is denied.
      ---- System.IO.IOException : Permission denied
      Stack Trace:
        /_/src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs(71,0): at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Func`4 createOpenException)
        /_/src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs(224,0): at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Permissions openPermissions, Int64& fileLength, Permissions& filePermissions, Func`4 createOpenException)
        /_/src/libraries/System.Private.CoreLib/src/System/IO/File.cs(171,0): at System.IO.File.OpenHandle(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
        /_/src/libraries/System.IO.FileSystem/tests/File/GetSetAttributes_SafeFileHandle.cs(10,0): at System.IO.Tests.GetSetAttributes_SafeFileHandle.GetAttributes(String path)
        /_/src/libraries/System.IO.FileSystem/tests/Base/FileGetSetAttributes.cs(49,0): at System.IO.Tests.FileGetSetAttributes.AssertSettingAttributes(String path, FileAttributes attributes)
           at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
        /_/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs(69,0): at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

Reply:
I guess this is what happens here:

`attributes: ReadOnly`

This sets the file to `ReadOnly`, and then:
```
        protected override FileAttributes GetAttributes(string path)
        {
            using var fileHandle =
                File.OpenHandle(path, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
            return File.GetAttributes(fileHandle);
        }
```
throws because it tries to open with `FileAccess.ReadWrite`.

So, changing `GetAttributes` to use `FileAccess.Read` should make the test work on Unix too.

Reply:
I had to make similar changes like for GetSetTimes - See `CanBeReadOnly`